build-logic/build-update-utils/src/main/kotlin/gradlebuild.wrapper.gradle.kts

```diff
@@ -1,11 +1,18 @@
 import gradlebuild.basics.capitalize
 import com.google.gson.Gson
 import java.net.URL
+import java.nio.charset.Charset
 
 wrapperUpdateTask("nightly", "nightly")
 wrapperUpdateTask("rc", "release-candidate")
 wrapperUpdateTask("current", "current")
 
+private
+fun Provider<RegularFile>.writeText(text: String, charset: Charset = Charsets.UTF_8) = this.get().asFile.writeText(text, charset)
+
+private
+fun Provider<RegularFile>.readText(charset: Charset = Charsets.UTF_8) = this.get().asFile.readText(charset)
```

Comment:
🤔 Out-of-scope for this PR: These should be standard functions. I wonder if we can/should introduce a `RegularFileProvider` interface, or if it's better to add these as Kotlin / Groovy DSL extension methods instead. But users shouldn't need to define these methods themselves.

Reply:
We've discussed that in larger group and came to the conclussion that such methods would be problematic, since users could accidentally eagerly evaluate a provider.

So it seems the general view is that we should not add such methods